### ISSUE TITLE
no-empty-interface and weak types

### ISSUE THREAD

**Issue by Crazometer** [palantir/tslint#3036]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: VS Code/node

#### TypeScript code being linted

```ts
interface Foo {
  blah()?: void;
}
class Bar {
}
interface Bar extends Foo { }
```

with `tslint.json` configuration: `no-empty-interface`

#### Discussion

As expected the code above triggers the `no-empty-interface` rule.

Prior to typescript 2.4 you could have the following:

```ts
interface Foo {
  blah()?: void;
}
class Bar implements Foo {
}
```
That is no longer allowed with the rationale that implementing `Foo` but none of the members is equivalent to doing nothing (_weak types_). The first block of code is the recommended solution.

The use case is `Foo` has a whole heap of members (~15) and I know some of them will be in `Bar` but I can't be sure which ones. 

Can we get an exception/option for the `no-empty-interface` rule that allows empty declarations if the interface will be merged?

There is some related discussion in:
https://github.com/ceolter/ag-grid/issues/1708
https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
